components/brave_ads/core/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -55,12 +56,20 @@ void TextEmbedding::Process(const std::string& html) {
     return;
   }
 
-  const ml::pipeline::EmbeddingProcessing* const embedding_proc_pipeline =
+  const ml::pipeline::EmbeddingProcessing* const processing_pipeline =
       resource_->Get();
   const ml::pipeline::TextEmbeddingInfo text_embedding =
-      embedding_proc_pipeline->EmbedText(text);
-  if (text_embedding.embedding.GetNonZeroElementCount() == 0) {
-    BLOG(1, "Failed to embed text");
+      processing_pipeline->EmbedText(text);
+  const std::vector<float> embedding = text_embedding.embedding;
```

Comment:
as I understand we don't need a copy here, could you please use `text_embedding.embedding` instead of `embedding`

Reply:
removed